src/main/java/io/vertx/ext/mongo/TimeSeriesOptions.java

```diff
@@ -0,0 +1,155 @@
+package io.vertx.ext.mongo;
+
+import com.mongodb.client.model.TimeSeriesGranularity;
+import com.mongodb.lang.Nullable;
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.codegen.annotations.GenIgnore;
+import io.vertx.core.json.JsonObject;
+import java.util.Objects;
+
+/**
+ * Options used to configure timeseries options.
+ *
+ * <p>Added in MongoDB 5.0 https://www.mongodb.com/docs/manual/core/timeseries-collections/
+ *
+ * @author <a href="mailto:mail@liuchong.io">Liu Chong</a>
+ */
+@DataObject(generateConverter = true)
+public class TimeSeriesOptions {
+  // required for time series collections
+  private String timeField;
+  private String metaField;
+  private TimeSeriesGranularity granularity;
```

Comment:
Can you please create an enum similar to `io.vertx.ext.mongo.WriteOption` or `io.vertx.ext.mongo.BulkOperationType` and use it here?

This would make the code more robust and avoid comparisons with uppercased strings.

Reply:
But `TimeSeriesGranularity` from mongo driver is already an enum, do you mean create our own enum to represent mongo driver's enum? something like:

```java
public enum TimeSeriesGranularity {
  SECONDS,
  MINUTES,
  HOURS;

  public com.mongodb.client.model.TimeSeriesGranularity getMongoDriverTimeSeriesGranularity() {
    return com.mongodb.client.model.TimeSeriesGranularity.valueOf(this.name());
  }
}
```